### PR TITLE
To resolve that memery leak result by deleting IP from interace with command 'dpip addr del' in snat mod

### PR DESCRIPTION
**Pull request by worldpeace365**
To resolve that memery leak result by deleting IP from interace with command 'dpip addr del' in snat mod

### CODE REVIEW

File: src/inetaddr.c [hunk 1]
```diff
@@ -472,18 +472,11 @@ struct inet_ifaddr *inet_addr_ifa_get_expired(int af, const struct netif_port *d
     struct inet_ifaddr *ifa;
     struct inet_device *idev;
 
-    if (!dev) {
-        dev = inet_addr_get_iface(af, addr);
-        if (!dev)
-            return NULL;
-    }
```
Comment (legend050709) [inline src/inetaddr.c:479]: conn超时的时候，conn中传递的dev不为NULL吧。怎么走进到这里的。

File: src/inetaddr.c [hunk 2]
```diff
@@ -472,18 +472,11 @@ struct inet_ifaddr *inet_addr_ifa_get_expired(int af, const struct netif_port *d
...
-
-    idev = dev_get_idev(dev);
-    assert(idev != NULL);
-
     ifa = expired_ifa_lookup(idev, addr, 0, af);
```
Comment (ywc689) [inline src/inetaddr.c:475]: `idev` is an uninitialized pointer now. I don't think the `expired_ifa_lookup` can succeed.

File: src/inetaddr.c [hunk 3]
```diff
@@ -472,18 +472,11 @@ struct inet_ifaddr *inet_addr_ifa_get_expired(int af, const struct netif_port *d
...
-
-    idev_put(idev);
+    if (ifa != NULL) {
+   	assert(ifa->idev != NULL);
+    	idev_put(ifa->idev);
```
Comment (ywc689) [inline src/inetaddr.c:478]: Why does it need `idev_put(ifa->idev)` here?

File: src/inetaddr.c [hunk 4]
```diff
@@ -769,7 +762,7 @@ static int ifa_expire(void *arg)
 
     err = inet_addr_del(ifa->af, ifa->idev->dev, &ifa->addr, ifa->plen);
     if (err != EDPVS_OK) {
-        RTE_LOG(ERR, IFA, "%s: inet_addr_del failed\n", __func__);
+        RTE_LOG(ERR, IFA, "inet_addr_del failed\n", __func__);
```
Comment (ywc689) [inline src/inetaddr.c:765]: The change is not correct here.

**Comment by worldpeace365** [comment]
[DPVS在SNAT场景下删除接口IP时大页内存泄漏.pdf](https://github.com/iqiyi/dpvs/files/10534204/DPVS.SNAT.IP.pdf)
企业微信截图_16750703461571
@ywc689 @legend050709 ，请二位看下文档里的说明，之前微信群跟开发人员也确认过是问题。而且多次复现过，必现。

File: src/inetaddr.c (hunk 1 above)
Comment (worldpeace365) [reply src/inetaddr.c:479]: > 
[DPVS在SNAT场景下删除接口IP时大页内存泄漏.pdf](https://github.com/iqiyi/dpvs/files/10534268/DPVS.SNAT.IP.pdf)

Comment (worldpeace365) [reply src/inetaddr.c:479]: 请看下上面这个pdf文件里说明了为什么为走到这里。

**Pull request closed without merging**